source/particles/particle_handler.cc

```diff
@@ -1221,7 +1224,13 @@ namespace Particles
           }
       }
 
-    send_recv_particles(ghost_particles_by_domain, ghost_particles);
+    if (enable_cache)
+      {
+        send_recv_cache_particles(ghost_particles_by_domain, ghost_particles);
```

Comment:
Maybe call it `send_recv_particles_and_build_cache`.

Reply:
Sure thing, makes sense!